### DISCUSSION TITLE
Problem preservating the Amiga version of "P-47" by Firebird

### DISCUSSION THREAD

**Discussion by realmml** [keirf/disk-utilities#137] [Q&A]
At the end of the Mission Elevator thread, Keir commented:
> I have 2 dumps of P-47 and will take a look. I know the ctraw image for both do not boot in WinUAE. The format is rainbird, but it appears that there is something special about the track 0 or both of my versions are bad. I will need to investigate track 0

I have two dumps of the game, in my case the track 0 is normal amigados, but if I try to convert with disk-analyze choosing a Rainbird format, I can not generate a bootable disk. This is the output:
```
$> disk utilities master>disk-analyse.exe -f "Midwinter" convert/p47a/track p47a.ipf
Side 0:
 T0: AmigaDOS
 T1-2: Rainbird
 T3-4: Unformatted*
 T5-37: Unformatted
 T38-78: Rainbird
 T79: Unformatted*
 T80-83: Unformatted
Side 1:
 T0-79: Rainbird
 T80-83: Unformatted
** WARNING: 3 tracks are damaged or unidentified!

$> disk-analyse.exe -f "Midwinter" convert/p47b/track p47b.ipf
Side 0:
 T0: AmigaDOS
 T1-2: Rainbird
 T3-4: Unformatted*
 T5-37: Unformatted
 T38-78: Rainbird
 T79: Unformatted*
 T80-83: Unformatted
Side 1:
 T0-79: Rainbird
 T80-83: Unformatted
** WARNING: 3 tracks are damaged or unidentified!
```
You can find my dumps here:
[P47 Dump A](https://www.dropbox.com/s/u37g2r2jo46sx0n/p47a.zip)
[P47 Dump B](https://www.dropbox.com/s/7t90hlsvlmno75t/p47b.zip)

**Comment by keirf** [comment]
> At the end of the Mission Elevator thread, Keir commented:
> 
> > I have 2 dumps of P-47 and will take a look. I know the ctraw image for both do not boot in WinUAE. The format is rainbird, but it appears that there is something special about the track 0 or both of my versions are bad. I will need to investigate track 0

That was Keith's comment.

**Reply by realmml** [reply]
Sorry, yes, it was Keith.

**Comment by kkrellwitz** [comment]
I will be getting to this shortly.  I am able to create a working version from both of your dumps I just need to better understand the protection track and recreate properly.

**Reply by realmml** [reply]
Thanks Keith! :)

**Comment by kkrellwitz** [comment]
@realmml This has turned out to be a bit more difficult than I expected.  Looks like Tracks 3.0 & 4.0 have a type of speedlock.  I am not familiar with speedlocks, so maybe @keirf can take a look at this.  In the mean time I will take a look at a decoder for a supported game that uses a speedlock and try to figure it out.

**Reply by keirf** [reply]
It is Speedlock. I can't see any twist to its use here on the protection tracks at all. I haven't looked at the protection code for P-47 however.

I did comment a Speedlock once, long ago https://github.com/keirf/amiga-stuff/blob/master/cracking/protections/speedlock_Dragons_Breath
Some of the comments are surely wrong (eg. Speedlock "sectors" aren't as long as I thought when commenting that disassembly).
